src/disk_interface.cc

```diff
@@ -100,19 +100,20 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
       static_cast<FINDEX_INFO_LEVELS>(1);
   FINDEX_INFO_LEVELS level =
       can_use_basic_info ? kFindExInfoBasic : FindExInfoStandard;
-  WIN32_FIND_DATAA ffd;
-  HANDLE find_handle = FindFirstFileExA((dir + "\\*").c_str(), level, &ffd,
-                                        FindExSearchNameMatch, NULL, 0);
+
+  WIN32_FIND_DATA ffd;
+  HANDLE find_handle = FindFirstFileEx(Utf8ToWide(dir + "\\*").c_str(), level,
+                                       &ffd, FindExSearchNameMatch, NULL, 0);
 
   if (find_handle == INVALID_HANDLE_VALUE) {
     DWORD win_err = GetLastError();
     if (win_err == ERROR_FILE_NOT_FOUND || win_err == ERROR_PATH_NOT_FOUND)
       return true;
-    *err = "FindFirstFileExA(" + dir + "): " + GetLastErrorString();
+    *err = "FindFirstFileEx(" + dir + "): " + GetLastErrorString();
     return false;
   }
   do {
-    string lowername = ffd.cFileName;
+    string lowername = WideToUtf8(ffd.cFileName);
     if (lowername == "..") {
```

Comment:
Don't convert prior to checking this if condition.

If the if check succeeds, then the result of your conversion get thrown away.

Convert after, where it'll *always* matter.